pymc/distributions/distribution.py

```diff
@@ -354,11 +355,266 @@ def dist(
+        graph, Symbolic distributions correspond to a larger graph that contains one or
+        more RandomVariables and an arbitrary number of deterministic operations, which
+        represent their own kind of distribution.
+
+        The graphs returned by symbolic distributions can be evaluated directly to
+        obtain valid draws and can further be parsed by Aeppl to derive the
+        corresponding logp at runtime.
+
+        Symbolic distributions must implement the following classmethods:
+        cls.dist
+            Performs input validation and converts optional alternative parametrizations
+            to a canonical parametrization. It should call `super().dist()` and pass
+            a list with the default parameters as the first and only non keyword argument,
+            followed by other keyword arguments like size and rngs.
+        cls.rv_op
+            Returns a TensorVariable that represents the symbolic distribution
+            parametrized by a default set of parameters and a size and rngs arguments
+        cls.ndim_supp
+            Returns the support of the symbolic distribution, given the default
+            parameters.This may not always be constant, for instance if the symbolic
+            distribution can be defined based on an arbitrary base distribution.
+        cls.change_size
+            Returns an equivalent symbolic distribution with a different size. This is
+            analogous to `pymc.aesarf.change_rv_size` for `RandomVariable`s.
+        cls.graph_rvs
+            Returns base RVs in a symbolic distribution.
+
+        Parameters
+        ----------
+        cls : type
+            A distribution class that inherits from SymbolicDistribution.
+        name : str
+            Name for the new model variable.
+        rngs : optional
+            Random number generator to use for the RandomVariable(s) in the graph.
+        dims : tuple, optional
+            A tuple of dimension names known to the model.
+        initval : optional
+            Numeric or symbolic untransformed initial value of matching shape,
+            or one of the following initial value strategies: "moment", "prior".
+            Depending on the sampler's settings, a random jitter may be added to numeric,
+            symbolic or moment-based initial values in the transformed space.
+        observed : optional
+            Observed data to be passed when registering the random variable in the model.
+            See ``Model.register_rv``.
+        total_size : float, optional
+            See ``Model.register_rv``.
+        transform : optional
+            See ``Model.register_rv``.
+        **kwargs
+            Keyword arguments that will be forwarded to ``.dist()``.
+            Most prominently: ``shape`` and ``size``
+
+        Returns
+        -------
+        var : TensorVariable
+            The created variable, registered in the Model.
+        """
```

Comment:
Doc string is super clear, a short example would really drive the point home.

Reply:
Alternatively can point to `Censored`.

Reply:
Added a reference to Censored